### ISSUE TITLE
Java Auto Completion only shows documentation in VSCode on Windows 10, but not on Linux Manjaro

### ISSUE THREAD

**Issue by GitFenixZ** [redhat-developer/vscode-java#2420]
##### Environment
- Operating System: Manjaro Linux x86_64
- Kernel: 5.17.1-3-MANJARO
- Desktop Environment: Plasma 5.24.4
- JDK version:
$ java --version
openjdk 17.0.3 2022-04-19
OpenJDK Runtime Environment (build 17.0.3+3)
OpenJDK 64-Bit Server VM (build 17.0.3+3, mixed mode)
$ javac --version
javac 17.0.3

- Visual Studio Code version:
$ code --version
1.66.2
dfd34e8260c270da74b5c2d86d61aee4b6d56977
x64 

- Java extension version: vscjava.vscode-java-pack@0.22.3

##### Steps To Reproduce

What I did beforehand:

1. I set my `JDK_HOME` and `PATH` environment variables in ~/.zshrc and ~/.bashrc 
```
export JDK_HOME=/usr/lib/jvm/default
export PATH=$JDK_HOME:$PATH
``` 
by following https://stackoverflow.com/a/663726 and adapting to my case.
 `/usr/lib/jvm/default` is a symlink to `/usr/lib/jvm/java-17-openjdk`. I used the symlink to keep `JDK_HOME` up to date when I install a new `openjdk` version.

2. I didn't specify the `java.jdt.ls.java.home` setting in VSCode's `settings.json` as the `Language Support for Java(TM) by Red Hat` extension page mentioned https://marketplace.visualstudio.com/items?itemName=redhat.java#java-tooling-jdk in order for the extension to find my `JDK_HOME` environment variable.

3. Relevant settings in `settings.json` (You can find the complete `settings.json` in `settings.txt` file below):
```
{
   "java.completion.enabled": true,
   "vsintellicode.java.completionsEnabled": true,
}
```
[settings.txt](https://github.com/redhat-developer/vscode-java/files/8514532/settings.txt)

4. Write java code (Sample code: `Test.java` below)
5. Call a function like `substring` on a String but write a part of the function to test the Auto Completion for example in our case: `sub`
6. **Hit Ctrl+Space** to show Auto Complete Suggestion
7. Once again **Hit Ctrl+Space** to develop the function documentation window.

_Sample code: Test.java_
```
import java.util.LinkedList;

public class Test {

    LinkedList<Integer> p = new LinkedList<>();
    public static void a(int g) {
        System.out.println();
        String s = "Hello!";
        String sub_s = s.sub
    }
}
```
_Logs: logfile.log_
[logfile.log](https://github.com/redhat-developer/vscode-java/files/8514310/logfile.log)

##### Current Result

What I get:
- For `substring` on Linux:
![Code_Substring_Linux_2022-04-19_15-27-52](https://user-images.githubusercontent.com/71153264/164037172-a84f5320-846a-4859-b943-9cc80545592c.png)

- For `println` on Linux:
![Code_Sysout_Linux_2022-04-19_15-28-30](https://user-images.githubusercontent.com/71153264/164037448-7b51ce47-4194-4f00-ade6-5a0a38813a81.png)

##### Expected Result
What I want: (the screenshots were taken from Windows to show what I wanted)
- For `substring` on Linux: 
![Code_Substring_Windows_wgSQInBv8h](https://user-images.githubusercontent.com/71153264/164037269-d680df43-8f8d-4c53-8ae1-77432c92fd92.png)

- For `println` on Linux:
![Code_Sysout_Windows_dj9j7zLjbp](https://user-images.githubusercontent.com/71153264/164042760-d4f3afd4-dcda-4c53-ba94-00390099145c.png)

**Comment by snjeza** [comment]
@GitFenixZ Could you try to install the JDK source code - java-17-openjdk-src?

**Comment by GitFenixZ** [comment]
@snjeza `java-17-openjdk-src` isn't available for me neither with `yay` nor `pacman` but I find
```
$ yay -Ss 17 openjdk src                                              
aur/openjdk-src-xdg 17.0.3.u3-1 (+0 0.00) 
    OpenJDK Java 17 sources - with improved Support for the XDG Base Directory Specification
aur/openj9-src 17.0.1.u12_openj9_0.29.1-1 (+3 0.26) 
    OpenJDK Java 17 sources
extra/openjdk-src 17.0.3.u3-1 (44.2 MiB 49.3 MiB) 
    OpenJDK Java 17 sources
```

Which one should I choose?

**Comment by fbricon** [comment]
@snjeza isn't JDT supposed to be able to fetch the Javadoc remotely, based on the JDK version?

**Comment by snjeza** [comment]
> @snjeza isn't JDT supposed to be able to fetch the Javadoc remotely, based on the JDK version?

JavadocContents.getMethodDoc(IMethod) returns null for String.substring(...) when using Javadoc remotely.

> java-17-openjdk-src isn't available for me neither with yay nor pacman but I find
...

You should check it with
```
ls /usr/lib/jvm/java-17-openjdk/lib/src.zip
```

**Comment by GitFenixZ** [comment]
@snjeza Thank you!

with
>  `ls /usr/lib/jvm/java-17-openjdk/lib/src.zip`
I realised I was missing `/usr/lib/jvm/java-17-openjdk/lib/src.zip`.

So I installed `openjdk-src 17.0.3.u3-1` with `yay -S openjdk-src 17.0.3.u3-1` and now it works I get the documentation in VSCode!

So if I understood my problem is that I needed the `OpenJDK Java 17 sources` as I use OpenJDK 17.

Now, my question is why didn't it get installed with `jdk-openjdk 17.0.3.u3-1` the `OpenJDK Java 17 development kit`? Shouldn't come with it?
